core/src/main/java/org/apache/accumulo/core/data/ArrayByteSequence.java

```diff
@@ -160,6 +160,10 @@ public int offset() {
    * @since 3.1.0
    */
   public void reset(byte[] data, int offset, int length) {
+    if (offset < 0 || offset > data.length || length < 0 || (offset + length) > data.length) {
+      throw new IllegalArgumentException(" Bad offset and/or length data.length = " + data.length
+          + " offset = " + offset + " length = " + length);
+    }
```

Comment:
@keith-turner identified [on another issue](https://github.com/apache/accumulo/pull/4524#discussion_r1692342340) that Java's built-in [Objects](https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/util/Objects.html) has some methods to check bounds as well.

The exceptions thrown for each of the 3 (what we have, Objects, Preconditions) are all slightly different, but are all RuntimeExceptions, so fundamentally they all should check things the same way... it would only matter if somebody had a specific catch block checking for which RTE it was.

Of the options, I'm inclined to choose [the built-in Objects version][1].

[1]: https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/util/Objects.html#checkFromIndexSize(int,int,int)

Reply:
Updated to use the Objects methods. They throw IndexOutOfBoundsException instead, which I updated in the unit tests in ArrayByteSequenceTest as well.